src/loaders/gtdb/parse_computed_genome_attribs.py

```diff
@@ -0,0 +1,223 @@
+PROTOTYPE
+
+This script involves processing computed genome attributes and organizing them into a structured format suitable
+for importing into ArangoDB. The resulting JSON file will be used to upsert the database with the parsed genome
+attribute data.
+
+Note: The database was previously created using a JSON file generated by the gtdb_genome_attribs_loader.py script.
+      It is important to ensure that the arguments "--load_ver" and "--kbase_collection" match the ones used in the
+      gtdb_genome_attribs_loader.py script in order to generate the same key for the corresponding Arango document.
+
+usage: parse_computed_genome_attribs.py [-h] --tools TOOLS [TOOLS ...] --load_ver LOAD_VER
+                                        [--kbase_collection KBASE_COLLECTION] [--root_dir ROOT_DIR] [-o OUTPUT]
+options:
+  -h, --help            show this help message and exit
+
+required named arguments:
+  --tools TOOLS [TOOLS ...]
+                        Extract results from tools. (e.g. gtdb_tk, checkm2, etc.)
+  --load_ver LOAD_VER   KBase load version (e.g. r207.kbase.1). This argument should be consistent with the one used by the
+                        gtdb_genome_attribs_loader.py script.
+
+optional arguments:
+  --kbase_collection KBASE_COLLECTION
+                        KBase collection identifier name (default: loader_common_names.DEFAULT_KBASE_COLL_NAME).
+                        This argument should be consistent with the one used by the gtdb_genome_attribs_loader.py script.
+  --root_dir ROOT_DIR   Root directory.
+  -o OUTPUT, --output OUTPUT
+                        output JSON file path (default: gtdb_computed_genome_attribs.json)
+
+"""
+import argparse
+import copy
+import os
+import sys
+
+import pandas as pd
+
+from src.loaders.common import loader_common_names
+from src.loaders.gtdb.gtdb_loader_helper import convert_to_json, init_genome_atrri_doc
+
+# Default result file name for parsed computed GTDB genome attributes data for arango import
+GTDB_COMPUTED_GENOME_ATTR_FILE = "gtdb_computed_genome_attribs.json"
+
+# The following features will be extracted from the CheckM2 result quality_report.tsv file as computed genome attributes
+SELECTED_CHECKM2_FEATURES = {'Completeness', 'Contamination'}
+
+# The following features will be extracted from the GTDB-TK summary file
+# ('gtdbtk.ar53.summary.tsv' or 'gtdbtk.bac120.summary.tsv') as computed genome attributes
+SELECTED_GTDBTK_SUMMARY_FEATURES = {'classification', 'fastani_taxonomy', 'closest_placement_taxonomy',
+                                    'pplacer_taxonomy'}
+
+NOTIFICATION = ' This argument should be consistent with the one used by the gtdb_genome_attribs_loader.py script.'
+
+
+def _locate_dir(root_dir, kbase_collection, load_ver, check_exists=False, tool=''):
+    result_dir = os.path.join(root_dir, loader_common_names.COLLECTION_DATA_DIR, kbase_collection, load_ver, tool)
+
+    if check_exists and not (os.path.exists(result_dir) and os.path.isdir(result_dir)):
```

Comment:
Something seems off here... if `result_dir` exists the 2nd clause will be false and the `if` statement will short circuit, right? So the `isdir` will never be executed?

On the other hand if `result_dir` doesn't exist then `isdir` will be executed, which doesn't seem to make sense either

Reply:
I didn't follow. `os.path.exists(result_dir) and os.path.isdir(result_dir)` - if `result_dir` exists, it will check `isdir`. If `result_dir` doesn't exist, it will just return `False` and ignore `isdir`.

Reply:
never mind, I misread the code